datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -814,14 +870,68 @@ pub fn binary(
     Ok(Arc::new(BinaryExpr::new(l, op, r)))
 }
 
+// TODO file a ticket with arrow-rs to include these kernels
```

Comment:
```suggestion
// When arrow-rs has these kernels, can remove this implementation
// see https://github.com/apache/arrow-rs/issues/842
```

Filed ticket in arrow https://github.com/apache/arrow-rs/issues/842